src/librustc_mir/borrow_check/nll/type_check/mod.rs

```diff
@@ -883,6 +891,23 @@ impl<'a, 'gcx, 'tcx> TypeChecker<'a, 'gcx, 'tcx> {
         )
     }
 
+    fn sub_types_or_anon(
```

Comment:
Is "anon" here supposed to be `opaque`?

Reply:
Yes, it should be.

Reply:
Anyway this PR is already merged, so best to open a new one with these changes.